### PR TITLE
Update documentation regarding connectionMatcher and connectionNormal…

### PR DESCRIPTION
**Pull request by jainam1995**
Update documentation as we have deprecate connectionMatcher.js and connectionNormalizer.js files.

### CODE REVIEW

**Comment by jainam1995** [comment]
files-overview
files-sequence

File: docs/api-reference.md [hunk 1]
```diff
@@ -229,6 +211,9 @@ Example:
 
 ## Deprecated API
 
+### connectionNormalizer.js and connectionMatcher.js
+These .js files are deprecated as of Tableau 2020.3. The <connection-normalizer> component can be added to the connectionResolver.tdr fie as shown in the connection-normalizer section above.
```
Comment (rosswbrown) [inline docs/api-reference.md:215]: Can you add the element/attribute names.  Technically the files could have been named anything before

File: docs/example.md [hunk 2]
```diff
@@ -68,10 +68,28 @@ The Connector Resolver file (.tdr) is optional. Tableau uses it to create a conn
...
+            <attr>port</attr>
+            <attr>dbname</attr>
+            <attr>username</attr>
+            <attr>password</attr>
+        </attribute-list>               
+    </required-attributes>
+</connection-normalizer>
+
+```
+`<setImpersonateAttributes/>` adds support for impersonate attributes and should be in every connector.
```
Comment (rosswbrown) [inline docs/example.md:89]: `setImpersonateAttributes` can be removed, its done automatically.  I believe samples are already updated.

File: docs/example.md [hunk 3]
```diff
@@ -82,7 +100,40 @@ If the attributes and their values are identical then the connections are consid
...
-![]({{ site.baseurl }}/assets/pce-tdr.png)
+```
+<tdr class='postgres_odbc'>
+    <connection-resolver>
+        <connection-builder>
+            // This is the component number 4 in the diagram above
+            <script file='connectionBuilder.js'/> 
+        </connection-builder>
+        <connection-properties>
+            // This is the component number 5 in the diagram above
```
Comment (rosswbrown) [inline docs/example.md:111]: Can you add JDBC only in the comment here too

**Review by rosswbrown — COMMENTED** [review]
We should remove the arrow and { related to the dialect in the images.  Visually it implies the dialect is used to connect.  There are narrow cases when this is true at an implementation level, however the dominate case is for query formation, which occurs after initial connection.  The description in 7 below I think articulates this well.

File: docs/api-reference.md (hunk 1 above)
Comment (jainam1995) [reply docs/api-reference.md]: Added

File: docs/example.md (hunk 2 above)
Comment (jainam1995) [reply docs/example.md]: Removed

File: docs/example.md (hunk 3 above)
Comment (jainam1995) [reply docs/example.md:109]: Added

**Comment by jainam1995** [comment]
> We should remove the arrow and { related to the dialect in the images. Visually it implies the dialect is used to connect. There are narrow cases when this is true at an implementation level, however the dominate case is for query formation, which occurs after initial connection. The description in 7 below I think articulates this well.

Updated the images. Adding screenshot of the new images in comments below

**Comment by jainam1995** [comment]
files-overview
files-sequence

**Comment by jainam1995** [comment]
Addressed @rosswbrown comments

File: docs/example.md [hunk 4]
```diff
@@ -185,7 +245,7 @@ For JDBC, search for <span style= "font-family: courier new">Connection URL</spa
 JDBCProtocol Connection URL: jdbc:postgresql://postgres:5342/TestV1?user=test&password=********
```
Comment (rosswbrown) [inline docs/example.md:245]:
```suggestion
JDBCProtocol Connection URL: jdbc:postgresql://postgres:5342/TestV1
```
This example pre-dated the propertiesBuilder.  user/pass are no longer passed on the url.

**Review by rosswbrown — APPROVED** [review]

File: docs/example.md (hunk 4 above)
Comment (jainam1995) [reply docs/example.md]: Fixed

**Pull request merged**